### DISCUSSION TITLE
Column format with livewire component?

### DISCUSSION THREAD

**Discussion by howdu** [rappasoft/laravel-livewire-tables#912] [Q&A]
Is there a way of adding a custom button without creating a blade view file?

```php
Column::make('Actions')
                ->excludeFromColumnSelect()
                ->view('tables.videos.actions')
```

```html
<x-jet-button
    wire:click="$emit('openModal', 'videos.edit', {{ json_encode([$row->id]) }})">
    <x-heroicon-m-pencil class="h-4 w-4 fill-current" />
</x-jet-button>
```

I tried using the `format` function but this doens't render anything apart from html.  
```php
->format(
    function($value, $row, Column $column) {
        return "<x-jet-button wire:click=\"\$emit('openModal', 'videos.edit', {{ json_encode([$row->id]) }})\"><x-heroicon-m-pencil class=\"h-4 w-4 fill-current\" /></x-jet-button>";
    }
)
```

**Comment by jadersbr** [comment]
did you add the html method at the end of your string?

`Column::make("ID", "id")
->sortable()
->excludeFromColumnSelect()
->searchable()
                    ->format(
                        fn($value, $row, Column $column) => "<a href='". route('users.edit', $row->id) ."'>" . $row->id . "</a>"
                    )
                    ->html(),`

**Reply by howdu** [reply]
Thanks yeah even with `html()` the components are not registered only html. 

<img width="1212" alt="Screenshot 2022-09-15 at 09 03 45" src="https://user-images.githubusercontent.com/533658/190349704-04a09d95-cd36-4a8c-9b3a-f44bd4fa80f6.png">
